Lib/linecache.py

```diff
@@ -182,6 +182,12 @@ def lazycache(filename, module_globals):
         loader = getattr(spec, 'loader', None)
         if loader is None:
             loader = module_globals.get('__loader__')
+        if not ((mod_file := module_globals.get('__file__', None)) and mod_file.endswith(filename)):
```

Comment:
```suggestion
        mod_file = module_globals.get('__file__')
        if not mod_file or not mod_file.endswith(filename):
```

This check could be done before fetching the spec, name and loader.

For other reviewers:  moving this check *after* `if name and get_source` fails to solve the issue.